# EDA French hospital data about Covid 19

# Library

In [2]:
install.packages("dplyr")
library(dplyr)


  There is a binary version available but the source version is later:
      binary source needs_compilation
dplyr  1.0.6  1.0.7              TRUE

  Binaries will be installed


Warning message:
"package 'dplyr' is in use and will not be installed"

In [4]:
install.packages("ggplot2")
library(ggplot2)


  There is a binary version available but the source version is later:
        binary source needs_compilation
ggplot2  3.3.3  3.3.5             FALSE



installing the source package 'ggplot2'



## Import data

In [7]:
df_incid_reg <- read.csv("data/data_hopital/covid-hospit-incid-reg-2022-01-09-19h06.csv", sep=";")
#### Changement du nom des colonnes
# names(data) <- c("X", "année", "prénom", "genre", "naissance")
# Affichage des 5 premières lignes
head(df_incid_reg,5)

jour       nomReg     numReg incid_rea
1 2020-03-19 Guadeloupe 1      0        
2 2020-03-20 Guadeloupe 1      0        
3 2020-03-21 Guadeloupe 1      0        
4 2020-03-22 Guadeloupe 1      5        
5 2020-03-23 Guadeloupe 1      0

In [ ]:
# df_incid_reg <- read.csv("data/data_hopital/covid-hospit-incid-reg-2022-01-09-19h06.csv", sep=";")
# #### Changement du nom des colonnes
# # names(data) <- c("X", "année", "prénom", "genre", "naissance")
# # Affichage des 5 premières lignes
# head(df_incid_reg,5)

In [9]:
df_hebdo_covid <- read.csv("data/data_hopital/donnees-hospitalieres-covid19-2022-01-09-19h06.csv", sep=";")
#### Changement du nom des colonnes
# names(data) <- c("X", "année", "prénom", "genre", "naissance")
# Affichage des 5 premières lignes
head(df_hebdo_covid,5)

dep sexe jour       hosp rea HospConv SSR_USLD autres rad dc
1 01  0    2020-03-18  2    0  NA       NA       NA      1   0
2 01  1    2020-03-18  1    0  NA       NA       NA      1   0
3 01  2    2020-03-18  1    0  NA       NA       NA      0   0
4 02  0    2020-03-18 41   10  NA       NA       NA     18  11
5 02  1    2020-03-18 19    4  NA       NA       NA     11   6

-------------------------------------

## Exploration

#### dep

In [17]:
deps <- df_hebdo_covid %>% count(dep)
head(deps,3)

dep n   
1 01  1989
2 02  1989
3 03  1989

In [76]:
nb_dep <- length(deps$dep)
nb_dep

[1] 102

In [20]:
deps$dep

[1] 01  02  03  04  05  06  07  08  09  10  11  12  13  14  15  16  17  18 
 [19] 19  21  22  23  24  25  26  27  28  29  2A  2B  30  31  32  33  34  35 
 [37] 36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53 
 [55] 54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71 
 [73] 72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89 
 [91] 90  91  92  93  94  95  971 972 973 974 976 978
102 Levels: 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 21 ... 978

In [ ]:
# On a bien tous les départements de la France (+Outre mers + dep 978)

#### sexe

In [22]:
sexes <- df_hebdo_covid %>% count(sexe)
head(sexes,5)

sexe n    
1 0    67626
2 1    67626
3 2    66963

In [27]:
sexes$percent <- (sexes$n*100)/sum(sexes$n)
sexes

sexe n     percent 
1 0    67626 33.44262
2 1    67626 33.44262
3 2    66963 33.11475

#### jour

In [28]:
jours <- df_hebdo_covid %>% count(jour)
head(jours,3)

jour       n  
1 2020-03-18 305
2 2020-03-19 305
3 2020-03-20 305

In [29]:
length(jours$jour)

[1] 663

In [77]:
lst_date <- max(as.Date(jours$jour))
lst_date

[1] "2022-01-09"

In [78]:
fst_date <- min(as.Date(jours$jour))
fst_date

[1] "2020-03-18"

In [ ]:
# Les observations sont donc entre le début du premier confinement et le 9 janvier 2022

#### hosp

In [35]:
hosps <- df_hebdo_covid %>% count(hosp)
head(hosps,4)

hosp n   
1 0    4983
2 1    4157
3 2    3932
4 3    3527

In [79]:
more_nb_hosp <- hosps %>% filter(hosps$hosp == max(hosps$hosp))
more_nb_hosp

hosp n
1 3281 1

In [80]:
more_freq_nb_hosp <- hosps %>% filter(hosps$n == max(hosps$n))
more_freq_nb_hosp

hosp n   
1 0    4983

#### rea

In [36]:
reas <- df_hebdo_covid %>% count(rea)
head(reas,4)

rea n    
1 0   29349
2 1   21422
3 2   16662
4 3   13373

In [82]:
more_nb_rea <- reas %>% filter(reas$rea == max(reas$rea))
more_nb_rea

rea n
1 855 1

In [83]:
more_freq_rea <- reas %>% filter(reas$n == max(reas$n))
more_freq_rea

rea n    
1 0   29349

#### HospConv

In [42]:
HospConvs <- df_hebdo_covid %>% count(HospConv)
head(HospConvs,6)

HospConv n   
1 0        2136
2 1        2112
3 2        2294
4 3        2796
5 4        2812
6 5        2637

In [54]:
HospConvs %>% filter(HospConvs$HospConv == max(HospConvs$HospConv,na.rm = TRUE))

HospConv n
1 1006     1

In [57]:
HospConvs %>% filter(HospConvs$n == max(filter(HospConvs,!is.na(HospConvs$HospConv)),na.rm = TRUE))

HospConv n   
1 4        2812

In [59]:
HospConvs %>% filter(HospConvs$n == max(HospConvs$n,na.rm = TRUE))

HospConv n     
1 NA       110105

#### SSR_USLD

In [63]:
SSR_USLDs <- df_hebdo_covid %>% count(SSR_USLD)
head(SSR_USLDs,6)

SSR_USLD n   
1 0        6738
2 1        5872
3 2        4624
4 3        3883
5 4        3979
6 5        3165

In [64]:
SSR_USLDs %>% filter(SSR_USLDs$SSR_USLD == max(SSR_USLDs$SSR_USLD,na.rm = TRUE))

SSR_USLD n
1 524      1

In [65]:
SSR_USLDs %>% filter(SSR_USLDs$n == max(filter(SSR_USLDs,!is.na(SSR_USLDs$SSR_USLD)),na.rm = TRUE))

SSR_USLD n   
1 0        6738

In [66]:
SSR_USLDs %>% filter(SSR_USLDs$n == max(SSR_USLDs$n,na.rm = TRUE))

SSR_USLD n     
1 NA       110105

#### autres

In [68]:
df_autres <- df_hebdo_covid %>% count(autres)
head(df_autres,6)

autres n    
1 0      51682
2 1      15329
3 2       7603
4 3       4297
5 4       3204
6 5       2351

In [69]:
df_autres %>% filter(df_autres$autres == max(df_autres$autres,na.rm = TRUE))

autres n
1 72     2

In [70]:
df_autres %>% filter(df_autres$n == max(filter(df_autres,!is.na(df_autres$autres)),na.rm = TRUE))

autres n    
1 0      51682

In [71]:
df_autres %>% filter(df_autres$n == max(df_autres$n,na.rm = TRUE))

autres n     
1 NA     110105

#### dc

In [72]:
dcs <- df_hebdo_covid %>% count(dc)
head(dcs,6)

dc n   
1 0  3753
2 1  1774
3 2  1013
4 3   913
5 4   834
6 5   986

In [84]:
more_nb_dc <- dcs %>% filter(dcs$dc == max(dcs$dc,na.rm = TRUE))
more_nb_dc

dc   n
1 5059 1

In [85]:
more_freq_dc <- dcs %>% filter(dcs$n == max(dcs$n,na.rm = TRUE))
more_freq_dc

dc n   
1 0  3753

### Brief of the simple EDA per column

The study period is between 18th March 2020 and 9th January 2022, that's more than 2years.

In [86]:
fst_date
lst_date

[1] "2020-03-18"

[1] "2022-01-09"

Datas show us that the highiest number of notified hospitalisation is 3281 people at the same day through all french deps. And, hopefully the more frenquently numbre of notified hospitalisation is 0 at the same day.

In [87]:
more_nb_hosp
more_freq_nb_hosp

hosp n
1 3281 1

hosp n   
1 0    4983